src/Raven.Server/Documents/Includes/IncludeTimeSeriesCommand.cs

```diff
@@ -37,33 +46,138 @@ public void Fill(Document document)
                 if (Results.ContainsKey(docId))
                     continue;
 
-                var rangeResults = GetTimeSeriesForDocument(docId, kvp.Value);
-
-                Results.Add(docId, rangeResults);
-
+                var count = kvp.Value.Count(x => x.Name == Constants.TimeSeries.All);
+                switch (count)
+                {
+                    case 0:
+                    {
+                        var rangeResults = GetTimeSeriesForDocument(docId, kvp.Value);
+                        Results.Add(docId, rangeResults);
+                        break;
+                    }
+                    case 1:
+                    {
+                        // get all ts names
+                        var timeSeriesNames = new List<string>();
+                        TimeSeriesHandler.GetTimesSeriesNames(document, timeSeriesNames);
+                        var arr = new HashSet<AbstractTimeSeriesRange>();
+                        switch (kvp.Value.First())
```

Comment:
why first?

Reply:
if there is a request for range on `@all_timeseries` we forbidding to use additional ts includes, so it always going to be single.